File: leetcode/c/58-length-of-last-word.c
/**
 * https://leetcode.com/problems/length-of-last-word/
 */

#include <string.h>

int lengthOfLastWord(char* s){
  int length = 0;
  int i = strlen(s) - 1;
  
  while (s[i] == ' ') {
    i--;
  }
  
  while (i >= 0 && s[i] != ' ') {
    length++;
    i--;
  }
  
  return length;
}

File: leetcode/javascript/58-length-of-last-word.js
/**
 * https://leetcode.com/problems/length-of-last-word/
 */

/**
 * @param {string} s
 * @return {number}
 */
 var lengthOfLastWord = function(s) {
  let length = 0;
  let i = s.length - 1;
  
  while (s[i] === ' ') {
    i--;
  }
  
  while (i >= 0 && s[i] !== ' ') {
    length++;
    i--;
  }
  
  return length;
};